### PR TITLE
feat: event decoding for evm verify worker set handler

### PR DESCRIPTION
**Pull request by sammy1991106**
## Todos

- [x] Unit tests
- [ ] Manual tests
- [ ] Documentation
- [ ] Connect epics/issues

### CODE REVIEW

File: ampd/src/handlers/evm_verify_worker_set.rs [hunk 1]
```diff
@@ -0,0 +1,124 @@
+use error_stack::ResultExt;
+use serde::Deserialize;
+
+use axelar_wasm_std::voting::PollID;
+use events_derive::try_from;
+
+use crate::types::{EVMAddress, Hash, TMAddress, U256};
+
+#[derive(Deserialize, Debug)]
+pub struct Operators {
```
Comment (cgorenflo) [inline ampd/src/handlers/evm_verify_worker_set.rs:10]: what's the difference between a worker and operator?

File: ampd/src/handlers/evm_verify_worker_set.rs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    pub weights: Vec<(EVMAddress, U256)>,
+    pub threshold: U256,
+}
+
+#[derive(Deserialize, Debug)]
+pub struct WorkerSet {
+    pub tx_id: Hash,
+    pub log_index: u64,
```
Comment (cgorenflo) [inline ampd/src/handlers/evm_verify_worker_set.rs:18]: what is this information here? is that for verifying that a worker set has been updated?

File: ampd/src/handlers/evm_verify_worker_set.rs (hunk 1 above)
Comment (sammy1991106) [reply ampd/src/handlers/evm_verify_worker_set.rs:10]: My understanding is that workers are anyone running external services (voting, signing, maybe others as well) and operators are those signer addresses set in the EVM gateway contract. I don't like these names either, but I'm following the cosmowasm contracts' events for now.

File: ampd/src/handlers/evm_verify_worker_set.rs [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    #[serde(rename = "_contract_address")]
+    contract_address: TMAddress,
+    worker_set: WorkerSet,
+    poll_id: PollID,
+    source_chain: connection_router::types::ChainName,
+    source_gateway_address: EVMAddress,
+    confirmation_height: u64,
+    expires_at: u64,
+    participants: Vec<TMAddress>,
+}
```
Comment (cgorenflo) [inline ampd/src/handlers/evm_verify_worker_set.rs:35]: do we need a completely separate event?

Comment (sammy1991106) [reply ampd/src/handlers/evm_verify_worker_set.rs:35]: Maybe we can unify with an enum later, but I'd prefer going with everything separate and then come back to refactor if possible.

File: ampd/src/handlers/evm_verify_worker_set.rs (hunk 2 above)
Comment (sammy1991106) [reply ampd/src/handlers/evm_verify_worker_set.rs:18]: This is the information for us to locate the EVM event and do comparison for voting.

Comment (cgorenflo) [reply ampd/src/handlers/evm_verify_worker_set.rs:18]: Maybe call it `WorkerSetConfirmation` or something like that? It's weird to call it just `WorkerSet` with that information

File: ampd/src/handlers/evm_verify_worker_set.rs (hunk 3 above)
Comment (cgorenflo) [reply ampd/src/handlers/evm_verify_worker_set.rs:35]: ok

**Review by haiyizxx — APPROVED** [review]

**Review by cgorenflo — APPROVED** [review]

**Pull request merged**